io/net/net_tools.py

```diff
@@ -0,0 +1,289 @@
+        tmpdir = data_dir.get_tmp_dir()
+        self.backup_dir = os.path.join(tmpdir, 'net-tools')
+        if not os.path.isdir(self.backup_dir):
+            os.makedirs(self.backup_dir)
+        # Get Hostname
+        output = process.run(cmd="hostname", shell=True, verbose=False)
+        if output.exit_status != 0:
+            self.error("Failed To get hostname\n")
+        else:
+            if output.stdout is None:
+                # set hostname if not set
+                if not process.system("hostname localhost.localdomain"):
+                    self.error("Failed to set hostname\n")
+            process.system("hostname > %s/HOSTNAME" % self.backup_dir,
+                           verbose=False,
+                           shell=True)
+
+    def test_hostname(self):
+        """
+        Verify the functionality of the hostname command
+        """
+        hostname_cmd = "hostname"
+        output = process.run(cmd=hostname_cmd, shell=True, verbose=False)
+        if output.exit_status != 0 and output.stdout is None:
+            self.error("unexpected response from hostname command")
+
+        # Verifying different options provided by hostname
+        options_to_verify = self.params.get('hostname_opt', default="f")
+        for option in options_to_verify:
+            self.log.info("Verifying the hostname -%s option" % option)
+            status = process.system("hostname -%s" % option, verbose=False,
+                                    shell=True)
+            if status != 0:
+                self.error(
+                    "unexpected response from hostname %s command" %
+                    option)
+
+        # Test for setting new hostname using -F option
+        process.system(
+            "echo 'myhost.my-domain' > %s/MYHOSTNAME" %
+            self.backup_dir, verbose=False, shell=True)
+        status = process.system("hostname -F %s/MYHOSTNAME" % self.backup_dir,
+                                verbose=False,
+                                shell=True)
+        if status != 0:
+            self.error("Failed to change the hostname")
+        else:
+            output = process.run(
+                cmd=hostname_cmd, shell=True, verbose=False)
+            if not (re.search("myhost.my-domain", output.stdout)):
+                self.error("unexpected response from hostname -F command and \
+                                            hostname -F didn't set hostname")
+        # Restore hostname
+        status = process.system("hostname -F %s/HOSTNAME" % self.backup_dir,
+                                verbose=False,
+                                shell=True)
+        if status != 0:
+            self.error(
+                "unexpected response trying to restore the hostname \
+                from HOSTNAME")
+
+    def test_ifconfig(self):
+        """
+        Verify the functionality of the ifconfig
+        """
+        output = process.run(cmd="ifconfig", shell=True, verbose=False)
```

Comment:
Again, no need for shell... Please go through all `process.*` commands and reconsider where shell is needed and where it's not. Also please add the `sudo=True` where root permissions are required (I strongly suggest running avocado as non-root with sudo configured, but even without it you can usually guess which commads requires it).